### ISSUE TITLE
Provide toString for events

### ISSUE THREAD

**Issue by ScoreUnder** [DV8FromTheWorld/JDA#311]
When initially writing a bot, it's useful to be able to see what events are running through the bot to be able to figure out what you need to act on.

Unfortunately, just printing the event out results in the default toString, which does not provide any useful information about the event.

It would be nice if they had a toString so that the events could be passed straight to `System.out.println` or similar for debugging and development purposes.

**Comment by MinnDevelopment** [comment]
Can you give a use case for this "debugging" that isn't covered by simply putting the logger on `TRACE`

```java
WebSocketClient.LOG.setLevel(SimpleLog.Level.TRACE);
```

**Comment by ScoreUnder** [comment]
That function is not discoverable. You have to know it exists to be able to use it, which is the kind of thing I'm trying to work around here. The more you can find out about the code without having to read the entire source code tree, the better.

That function itself only provides part of the story. It does not say which fields in the event contain which values, and it does not say which event it is firing. It might be useful for inspecting the data given to us by the server, but it is not useful for quickly exploring which events are relevant to your bot or for clarifying the structure of the Java objects used, both of which are invaluable in the earlier stages of bot development.

**Comment by MinnDevelopment** [comment]
Why do you have to read the source tree to figure out what events exist and what they provide?
We have a pretty simple event hierarchy which is entirely [documented](http://home.dv8tion.net:8080/job/JDA/javadoc/) with package docs and class level docs, each method is listed and in a logical place to find.
I am not sure how printing every event to console (which is a MASSIVE spam btw) can be helpful for new users in any way and it is certainly a very big time investment to implement this.

Even the wiki has all events listed here: https://github.com/DV8FromTheWorld/JDA/wiki/8%29-List-of-Events

**Comment by ScoreUnder** [comment]
It's not massive spam. If you invite a small fledgling bot to a private test server, printing out every event to the console is fine. But more importantly, it precisely points you to the events you're looking for. Rather than having to search the full list of event types, or trying to guess what the official Discord API terminology for the event you're looking for is, you just look at what your bot printed and you will know what your bot needs to handle.

**Comment by MinnDevelopment** [comment]
I recommend if you want to learn more about events to just [join our discord](https://discord.gg/0hMr4ce0tIl3SLv5) and ask for the event there. This is not going to be implemented as it is a big task for a little payoff. I hope you can understand that.

**Comment by MinnDevelopment** [comment]
If you _really_ insist on printing every event into your console you can try using reflections to access all getters
```java
@Override
public synchronized void onGenericEvent(Event event)
{
    System.out.println("EventType: " + event.getClass().getSimpleName());
    for (Method m : event.getClass().getMethods())
    {
        final String name = m.getName();
        final Class<?> returnType = m.getReturnType();
        if ((name.startsWith("is") || name.startsWith("get")) && !returnType.equals(Void.class))
            System.out.println("\t" + name + ": " + returnType.getSimpleName());
    }
}
```
But honestly I think asking for events is much better and you will have a much better experience than trying to find it this way.

**Comment by MinnDevelopment** [comment]
Closing this as it seems inactive and I made my statement.

**Comment by ScoreUnder** [comment]
What kind of activity are you expecting?

**Comment by MinnDevelopment** [comment]
A single response to my two comments I made 6 days ago would be a start.

**Comment by ScoreUnder** [comment]
Fair enough. For the first, I don't see much of a problem with automatically generating the toString method. It doesn't need to get anything special across, just give an idea of the events being sent and the breadth of information that each specific event contains (for example, it's not intuitively obvious that the event for joining a voice chat does not include the muted/deafened state). IntelliJ can do most of the work with alt+insert.

Regarding reflection, that works to an extent but everyone who wants to try the same will either have to write the code from scratch or somehow come across this issue to find that code. The existence of reflection (let alone how to use it) will likely not be known to Java newbies, which rules out the first in arguably a majority of cases.

**Comment by MinnDevelopment** [comment]
I'm afraid that java newbies won't even understand how the event system works let alone that every event is a subclass of `Event`.
I doubt that the first thing a new JDA user would do would be implementing an `EventListener` and just doing `System.out.println(event);`. The first thing you would/should do is look at the list of events in the javadocs or in `ListenerAdapter` or in the wiki or just ask in the development discord. `ListenerAdapter` is explicitly mentioned in the `README` as well.

We have bigger priorities than adding a `toString` implementation to 140 events (client events excluded) such as fixing issues with permissions, moving towards etf or adding implementations for new features (audit logs)/client features (relationships).
My suggestion is that if you want this to be added you will have to do it yourself as our staff is not likely gonna do this any time soon.

**Comment by ScoreUnder** [comment]
>I doubt that the first thing a new JDA user would do would be implementing an `EventListener` and just doing `System.out.println(event);`. 

Then brace yourself, because that's exactly what I did. As for `ListenerAdapter``, I have still yet to find a practical use for it.

If this is a problem of priorities, then why not leave the ticket open? Closing it implies it is complete, or never to be completed. If you close a ticket, that says to any reader that you will not even be accepting pull requests to that effect.
